aten/src/ATen/native/ReduceOpsUtils.h

```diff
@@ -73,18 +73,19 @@ inline bool _dimreduce_return_trivial(Tensor &result, const Tensor &self,
   return false;
 }
 
-inline bool _dimreduce_return_trivial_no_ident(Tensor &result, const Tensor &self,
-                                               int64_t dim, bool keepdim, const char *fn_name) {
+inline bool _dimreduce_return_trivial_no_ident(Tensor &result, const Tensor &self, 
+                                               const char *fn_name) {
   if (self.numel() == 1 && self.ndimension() == 0) {
     result.resize_({});
     result.fill_(self);
     return true;
-  }
-
-  if (self.numel() == 0) {
+  } else if (self.numel() == 0 && self.ndimension() == 2) {
+    return false;
+  } else if (self.numel() == 0) {
```

Comment:
kalau di dalam statement `if` ada statement `return`, fungsi akan selesai jika masuk conditional `if` tersebut, jadi tidak perlu lagi ada `else`.

line 82 pada conditional if `(self.numel() == 0 && self.ndimension() == 2)` dalamnya `return false` (line 83) sama dengan return fungsi terakhir di line 89. Juga, line 82 dan 84 sama-sama check `self.numel() == 0`. Jadi, lebih baik kalau yang diganti hanya conditional di line 84 menjadi `(self.numel() == 0 && self.ndimension() != 2)`

Reply:
nice, thank you :ok_hand: